client/ayon_core/plugins/publish/extract_review.py

```diff
@@ -881,49 +972,184 @@ def ffmpeg_full_args(
+        current_repre: Dict[Any, Any],
+        start_frame: int,
+        end_frame: int
+    ) -> Union[Dict[int, str], None]:
+        """Tries to replace missing frames from ones from last version"""
+        repre_file_paths = self._get_last_version_files(
+            instance, current_repre)
+        if repre_file_paths is None:
+            # issues in getting last version files, falling back
+            return None
+
+        prev_collection = clique.assemble(
+            repre_file_paths,
+            patterns=[clique.PATTERNS["frames"]],
+            minimum_items=1
+        )[0][0]
+        prev_col_format = prev_collection.format("{head}{padding}{tail}")
+
+        added_files = {}
+        anatomy = instance.context.data["anatomy"]
+        col_format = collection.format("{head}{padding}{tail}")
+        for frame in range(start_frame, end_frame + 1):
+            if frame in collection.indexes:
+                continue
+            hole_fpath = os.path.join(staging_dir, col_format % frame)
+
+            previous_version_path = prev_col_format % frame
+            # limits too large padding coming from Anatomy
+            previous_version_path = (
+                os.path.join(
+                    anatomy.fill_root(os.path.dirname(previous_version_path)),
+                    os.path.basename(previous_version_path)
+                )
+            )
+            if not os.path.exists(previous_version_path):
+                self.log.warning(
+                    "Missing frame should be replaced from "
+                    f"'{previous_version_path}' but that doesn't exist. "
+                    "Falling back to filling from currently last rendered."
+                )
+                return None
+
+            self.log.warning(
+                f"Replacing missing '{hole_fpath}' with "
+                f"'{previous_version_path}'"
+            )
+            speedcopy.copyfile(previous_version_path, hole_fpath)
+            added_files[frame] = hole_fpath
+
+        return added_files
+
+    def _get_last_version_files(
+        self,
+        instance: pyblish.plugin.Instance,
+        current_repre: Dict[Any, Any],
+    ):
+        product_name = instance.data["productName"]
+        project_name = instance.data["projectEntity"]["name"]
+        folder_entity = instance.data["folderEntity"]
+
+        version_entity = get_last_version_by_product_name(
+            project_name,
+            product_name,
+            folder_entity["id"],
+            fields={"id"}
+        )
+        if not version_entity:
+            return None
+
+        repres = get_representations(
+            project_name,
+            version_ids=[version_entity["id"]]
+        )
+        matching_repre = None
+        for repre in repres:
+            if repre["name"] == current_repre["name"]:
+                matching_repre = repre
+                break
+        if not matching_repre:
+            return None
+
+        repre_file_paths = [
```

Comment:
The files won't have filled root, is that ok?

Reply:
Yes, its filled later.